tensorflow/python/distribute/input_lib.py

```diff
@@ -1014,17 +1373,23 @@ def _create_iterators_per_worker(worker_datasets, input_workers):
   for i, worker in enumerate(input_workers.worker_devices):
     with ops.device(worker):
       worker_devices = input_workers.compute_devices_for_worker(i)
-      iterator = _SingleWorkerDatasetIterator(worker_datasets[i], worker,
-                                              worker_devices)
+      if tf2.enabled() and not enable_legacy_iterators:
```

Comment:
You need to `from tensorflow.python import tf2` for this.

Reply:
Yep, working on another CP for this. Thanks!